lstchain/scripts/lstchain_cherenkov_transparency.py

```diff
@@ -0,0 +1,680 @@
+        return -2, -2, -2  # No maximum found
+
+    peak_values = y[indices_maxima]
+    peak_positions = x[indices_maxima]
+
+    # Exclude too-low peaks (may be just fluctuations if stats is low, e.g. at
+    # high intensities). If there are no peaks above miny, just return the
+    # height and position of the maximum one:
+    if peak_values.max() < miny:
+        return -3, peak_values.max(), peak_positions[np.argmax(peak_values)]
+
+    # Keep only those higher than miny:
+    above_miny = peak_values >= miny
+    peak_values = peak_values[above_miny]
+    peak_positions = peak_positions[above_miny]
+
+    # Peak values ordered from biggest to smallest:
+    ordered_peaks = np.sort(peak_values)[::-1]
+    ordered_peaks_pos = peak_positions[np.argsort(peak_values)][::-1]
+
+    # We first assume that the highest peak is the genuine cosmics peak.
+    # If x50 is too low to be a reasonable threshold for cosmics, perhaps
+    # the highest peak is an artifact (sometimes we see spurious peaks not due
+    # to showers, at low intensities - perhaps caused by meteors or satellites)
+    # In that case we try the next peaks in order of height, until we get
+    # one for which x50 is below min_intensity. Note: this does not work
+    # against fake peaks at high intensities, that we sometimes have e.g. due
+    # to the MAGIC LIDAR crossing the LST FOV. In those cases we may get a
+    # completely wrong value for x50 (i.e. unrelated to the cosmics intensity
+    # spectrum).
+
+    ipeak = -1
+    x50 = np.nan
+
+    while not (x50 >= min_intensity):  # Note this is true for x50==np.nan
+        ipeak += 1
+        if ipeak >= len(peak_values):
+            break
+
+        peak_bin = np.where(x == ordered_peaks_pos[ipeak])[0][0]
+        # We move from the peak towards left (lower intensity), until we find
+        # the 50% of the peak rate
+        x_left_side = x[:peak_bin]
+        y_left_side = y[:peak_bin]
+        # Get the last intensity before the peak for which the rate is 50% of
+        # peak:
+        ix50 = np.where(y_left_side < 0.5 * ordered_peaks[ipeak])[0]
+        if len(ix50) > 0:
+            x50 = x_left_side[ix50[-1]]
+        else:
+            x50 = np.nan
+
+    if not (x50 >= min_intensity):  # = did not find an x50 in a reasonable
+        # range
+        return -4, ordered_peaks[0], ordered_peaks_pos[0]
+        # in this case return dR/dI and intensity for the highest peak
+
+    return x50, ordered_peaks[ipeak], ordered_peaks_pos[ipeak]
+
+
+def pol1(x, a, b):
+    return a + b * x
+
+
+def pol2(x, a, b, c):
+    return a + b * x + c * x * x
+
+
+def expfunc(x, a, b):
+    return a * np.exp(b * x)
+
+
+def time_to_yyyymmdd(t):
+    hour = np.array([datetime.fromtimestamp(x).hour for x in t])
+    t2 = np.array([x if h > 12 else x - 43200 for x, h in zip(t, hour)])
+    dt = [datetime.fromtimestamp(x) for x in t2]
+
+    return np.array([(x.year * 10000 + x.month * 100 + x.day) for x in dt])
```

Comment:
I think you don't need this function, you already know the date in `YYYYMMDD` format from the beginning `date = int(output_file[-11:-3])`

Reply:
With the latest changes we do need it - and I prefer not to depend on a file name for that.